src/middleware.test.ts

```diff
@@ -29,4 +32,54 @@ describe("middleware.test.ts", () => {
       expect(mockHandlerFn).toHaveBeenCalledOnce();
     });
   });
+
+  describe("withRegisteredSaleorDomainHeader", () => {
+    let mockHandlerFn: Handler = vi.fn(getMockEmptyResponse);
+    const mockAPL: APL = {
+      get: async (domain: string) =>
+        domain === "example.com"
+          ? {
+              domain: "example.com",
+              token: "mock-token",
+            }
+          : undefined,
+      set: vi.fn(),
+      delete: vi.fn(),
+      getAll: vi.fn(),
+    };
+
+    beforeEach(() => {
+      mockHandlerFn = vi.fn();
+    });
+
+    it("Pass request when auth data are available", async () => {
+      const mockRequest = {
+        context: {},
+        headers: {
+          host: "my-saleor-env.saleor.cloud",
+          "x-forwarded-proto": "https",
+          [SALEOR_DOMAIN_HEADER]: "example.com",
+        },
+      } as unknown as Request;
+
+      await withRegisteredSaleorDomainHeader({ apl: mockAPL })(mockHandlerFn)(mockRequest);
+
+      expect(mockHandlerFn).toHaveBeenCalledOnce();
+    });
+
+    it("Reject request when auth data are not available", async () => {
+      const mockRequest = {
+        context: {},
+        headers: {
+          host: "my-saleor-env.saleor.cloud",
+          "x-forwarded-proto": "https",
+          [SALEOR_DOMAIN_HEADER]: "not-registered.example.com",
+        },
+      } as unknown as Request;
+
+      await withRegisteredSaleorDomainHeader({ apl: mockAPL })(mockHandlerFn)(mockRequest);
```

Comment:
I think you can actually assert if error response was returned - this is client facing stuff